fetch.bs

```diff
@@ -6655,6 +6697,239 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+
+  <ol>
+   <li><p>The user agent should wait until <var>activateAfter</var> milliseconds have passed,
+   or until the user agent has a reason to believe that it is about to lose the opportunity to
+   execute scripts, e.g., when the browser is moved to the background, or when <var>request</var>'s
+   <a for=request>client</a> is a <code>Document</code> that had a "<code>hidden</code>"
+   <a for=Document>visibility state</a> for a long period of time.
+
+   <li><p>The user agent may wait for a further <a>implementation-defined</a> <a>duration</a>, e.g.,
+   in order to batch several requests together or to wait until <a for=request>keepalive</a>
+   requests are complete.
+
+   <li><p><a>Process a deferred fetch</a> given <var>deferredRecord</var>.
+  </ol>
+ </li>
+
+ <li><p>Return <var>deferredRecord</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To get the <dfn>available deferred fetching quota</dfn> given a <code>Document</code>
+<var>document</var> and an <a for=/>origin</a>-or-null <var>origin</var>:
+
+<p class=note>This algorithm asserts that this deferred fetch doesn't exceed two quotas: one for the
+top-level document (640 kibibytes), and one for the reporting origin (64 kibibytes). The larger
+quota ensures that the top-level {{Document}} and its subresources don't continue using an unlimited
+amount of bandwidth after being destroyed. The smaller quota ensures that a single reporting sink
+doesn't reserve the whole quota to itself.
+
+<ol>
+ <li><p>Let <var>closestSameOriginInclusiveAncestor</var> be <var>document</var>'s
+ <a>node navigable</a>.
+
+ <li><p>While <var>closestSameOriginInclusiveAncestor</var>'s <a for=navigable>parent</a> is a
+ <a for=/>navigable</a> whose <a>active document</a>'s <a for=Document>origin</a> is <a>same origin</a>
+ with <var>document</var>'s <a for=Document>origin</a>, set
+ <var>closestSameOriginInclusiveAncestor</var> to <var>closestSameOriginInclusiveAncestor</var>'s
+ <a for=navigable>parent</a>.
+
+ <li><p>Let <var>directRelativeSameOriginNavigables</var> be the
+ <a>inclusive direct same origin descendants</a> of <var>closestSameOriginInclusiveAncestor</var>.
+
+ <li><p>Let <var>remainingTotalQuota</var> be 640 kibibytes if
+ <var>directRelativeSameOriginNavigables</var> includes <var>document</var>'s
+ <a>node navigable</a>'s <a for=navigable>top-level traversable</a>, and 64 kibibytes otherwise.
+
+ <li><p>Let <var>remainingQuotaForOrigin</var> be 64 kibibytes.
+
+ <li>
+  <p><a for=list>For each</a> <var>navigable</var> in
+  <var>directRelativeSameOriginNavigables</var>:
+
+  <ol>
+   <li>
+    <p><a for=list>For each</a> <a>deferred fetch record</a> <var>deferredRecord</var> in
+    <var>navigable</var>'s <a>active document</a>'s <a for=fetch>fetch group</a>'s
+    <a for="fetch group">deferred fetch records</a>:</p>
+
+    <ol>
+     <li><p>Let <var>length</var> be <var>deferredRecord</var>'s
+     <a for="deferred fetch record">request</a>'s <a for=request>body</a>'s
```

Comment:
> Let length be deferredRecord’s [request](https://whatpr.org/fetch/1647.html#deferred-fetch-record-request)’s [body](https://whatpr.org/fetch/1647.html#concept-request-body)’s [length](https://whatpr.org/fetch/1647.html#concept-body-total-bytes).

This step should also be updated to count URL and header's sizes (the following steps)?

> Let totalRequestLength be the [length](https://infra.spec.whatwg.org/#string-length) of request’s [URL](https://whatpr.org/fetch/1647.html#concept-request-url), [serialized](https://url.spec.whatwg.org/#concept-url-serializer) with [exclude fragment](https://url.spec.whatwg.org/#url-serializer-exclude-fragment) set to true.

> [For each](https://infra.spec.whatwg.org/#list-iterate) (name, value) in [header list](https://whatpr.org/fetch/1647.html#concept-request-header-list), increment totalRequestLength by name’s [length](https://infra.spec.whatwg.org/#byte-sequence-length) + value’s [length](https://infra.spec.whatwg.org/#byte-sequence-length).

Reply:
Fixed